internal/appsec/waf.go

```diff
@@ -184,7 +202,13 @@ func newGRPCWAFEventListener(handle *waf.Handle, _ []string, timeout time.Durati
 			if md := handlerArgs.Metadata; len(md) > 0 {
 				values[grpcServerRequestMetadata] = md
 			}
+			now := time.Now()
 			event := runWAF(wafCtx, values, timeout)
+			// WAF run durations are WAF context bound. As of now we need to keep track of those externally since
+			// we use a new WAF context for each callback. When we are able to re-use the same WAF context across
+			// callbacks, we can get rid of these variables and simply use the WAF bindings in OnHandlerOperationFinish.
+			wafRunDuration.Add(wafCtx.TotalRuntime())
+			wafBindingsRunDuration.Add(uint64(time.Since(now).Nanoseconds()))
```

Comment:
To be a bit more precise with `time.Since(now)`, please move it right after the call to `runWAF()` 🙏